Files liferea-1.4.16b.orig/.aclocal.m4.swp and liferea-1.4.16b/.aclocal.m4.swp differ
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN liferea-1.4.16b.orig/configure.ac liferea-1.4.16b/configure.ac
--- liferea-1.4.16b.orig/configure.ac	2008-06-11 18:08:41.000000000 -0400
+++ liferea-1.4.16b/configure.ac	2008-06-20 10:59:28.000000000 -0400
@@ -23,6 +23,8 @@ AC_ISC_POSIX
 AC_PROG_CC
 AM_PROG_CC_STDC
 AM_PROG_CC_C_O
+dnl Mozilla needs C++; don't make conditional, or autoconf fails
+AC_PROG_CXX
 AC_HEADER_STDC
 AC_PROG_INSTALL
 AC_PROG_LN_S
@@ -168,7 +170,7 @@ dnl *********
 if test "x$enable_xulrunner" = "xyes" ; then
  
    AC_MSG_CHECKING([for XulRunner 1.9+ support])  
-   PKG_CHECK_MODULES(XULRUNNER, libxul-embedding, XULRUNNER_PROVIDER=libxul-embedding, XULRUNNER_PROVIDER=)
+   PKG_CHECK_MODULES(XULRUNNER, libxul-embedding-unstable, XULRUNNER_PROVIDER=libxul-embedding-unstable, XULRUNNER_PROVIDER=)
    
    if test "x$XULRUNNER_PROVIDER" = "x" ; then
       AC_MSG_CHECKING([for XulRunner 1.8 support])  
@@ -186,9 +188,6 @@ if test "x$enable_xulrunner" = "xyes" ; 
       AC_SUBST(XULRUNNER_LIBS)
 
       MOZILLA_LIB_ROOT=`$PKG_CONFIG --libs-only-L $XULRUNNER_PROVIDER | awk '{print $1}' |  cut -c 3-`
-   
-      dnl the Mozilla XPCOM access means C++
-      AC_PROG_CXX
    fi
 fi
 
@@ -248,9 +247,6 @@ if test "x$enable_mozilla" = "xyes" ; th
 
    dnl AC_MSG_RESULT([$gecko_provider found!])
 
-   dnl the Mozilla XPCOM access means C++
-   AC_PROG_CXX
-   
    dnl Note: with the infos from mozilla-gtkmozembed.pc
    dnl we got all includes and libraries for the C interface
    dnl to Mozilla, but not the XPCOM headers we also need.
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN liferea-1.4.16b.orig/src/mozilla/xulrunner.c liferea-1.4.16b/src/mozilla/xulrunner.c
--- liferea-1.4.16b.orig/src/mozilla/xulrunner.c	2007-07-23 13:56:53.000000000 -0400
+++ liferea-1.4.16b/src/mozilla/xulrunner.c	2008-06-20 10:50:59.000000000 -0400
@@ -30,10 +30,12 @@
 
 static void xulrunner_init(void) {
 	
+#ifndef XPCOM_GLUE
 	/* gtk_moz_embed_set_comp_path() must not be called
 	   without a component path for XulRunner. Passing
 	   NULL will crash Liferea. */
 	gtk_moz_embed_set_comp_path(XULRUNNER_HOME);
+#endif
 
 	mozembed_init();	
 }
